### ISSUE TITLE
Question about TIMIT_labels.npy and others

### ISSUE THREAD

**Issue by andriuslauraitis** [mravanelli/SincNet#41]
Hello,
First of all, thanks for sharing your work :) 
I am new in speech recognition science field.

Wanted to ask couple of questions:

1) TIMIT_labels.npy. Is there a way to track how id's (particular numbers) are assigned in the dictionary? I loaded the file from python command prompt. For example  : 'train/dr5/fjxm0/sx311.wav': 267. How is 267 assigned (maybe there is a reference somewhere in TIMIT dataset)? As i understand the id in such case is: fjxm0.  So, is 267 important? Or, for example (all occurrences of 267 in the file can be changed to some new value).

2) Can you give some guidelines about how to choose training and test sets for the model? E.g. what percentage for training, and testing?

3) File: model_raw.pkl. Is the the file for trained model? How can i use this file?

4) As i understand SincNet solves 'speaker identification task'. What are the differences, compared to speech recognition task? How can i adapt Sincet for individual speaker? (e.g. to compare results for the same speaker spoken twice) Maybe for individual speaker, i need to design test set in a way that only audio files for that speaker is included?

Looking forward to hearing from you :)

Best regards,
Andrius L.

**Comment by mravanelli** [comment]
Hi,
thank you very much for your interest in my work!

1- What is really important is to assign a different integer for each
different speaker. The integer is a unique identifier for that speaker.
2- This really depends on the dataset and on the constraints that you have
in a practical application. To make the task more challenging you should
select only few seconds of traning material (in our case we use 10-15
second of training material for each speaker).
3- Yes, the final model is saved in model_raw.pkl. It contains all the
learned parameters of the neural network. You can load it using the
standard functions of pytorch for loading the saved model (see pytorch
tutorial).
4- In this repository SincNet is used for speaker-identification, but it is
really a general paradigm for processing audio sequences. In
https://github.com/mravanelli/pytorch-kaldi you can find SincNet applied to
speech recognition. I'm not sure to have understood what do you mean with
"How can i adapt Sincet for individual speaker".

Best,

Mirco



<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by andriuslauraitis** [comment]
Thanks for quick answer.

Regarding "How can i adapt Sincet for individual speaker":

For example i have such metrics (on trained SincNet model): 
epoch 360, loss_tr=0.033095 err_tr=0.009600 loss_te=4.254683 err_te=0.419954 err_te_snt=0.005772 

So the value  err_te_snt=0.005772 indicates "the classification error (measured at sentence level) of the test data" for the trained model. And test data contains (in TIMIT example) more than ~1300 examples.

In my case i have 20 speakers (more that 100 records): 10 healthy people and other 10 sick people (have minor problems with their voice). So i should put all audio files of sick people to test data? 

Maybe another way to ask would be like this: suppose i have a trained SincNet model (for example i take 19 persons and divide audio to train and test data accordingly). 

Then i want to make predictions of error metrics to new data samples (lets say for the 20-th person). Is Sincnet capable of doing that? Or this scenario is a "out of scope"?

Maybe there are some Sincnet comands that can  process audio sequence (e.g. on new data audio sample that is not present in train and test sets) using already trained model?  

My goal here is to try evaluate the same speaker (e.g. spoke defined sentence once (got and error metric e.g. err_te_snt_1), spoke the same sentence once more and got error metric err_te_snt_2).

Sorry for confusing questions :)

Best regards,
Andrius L.

**Comment by mravanelli** [comment]
Hi Andrius,
you can find more information about adding more speakers here:
https://github.com/mravanelli/SincNet/issues/37

hope this will be helpful for you..
